api/cpp/tests/datastructures.cpp

```diff
@@ -39,6 +39,12 @@ SCENARIO("SharedString API")
         REQUIRE(str == "Hello 🦊!");
         REQUIRE(std::string_view(str.data()) == "Hello 🦊!");
     }
+
+    SECTION("begin/end")
+    {
+        str = "Hello";
+        REQUIRE(str.begin() != str.end());
```

Comment:
Maybe the test can actually test even more:
```c++
        REQUIRE(str.begin() + str.size() == str.end());
```

Reply:
Good idea.